jsonrpc/endpoints_debug.go

```diff
@@ -119,6 +129,131 @@ func (d *DebugEndpoints) TraceBlockByHash(hash types.ArgHash, cfg *traceConfig)
 	})
 }
 
+// TraceBatchByNumber creates a response for debug_traceBatchByNumber request.
+// this endpoint tries to help clients to get traces at once for all the transactions
+// attached to the same batch.
+//
+// IMPORTANT: in order to take advantage of the infrastructure automatically scaling,
+// instead of parallelizing the trace transaction internally and pushing all the load
+// to a single jRPC and Executor instance, the code will redirect the trace transaction
+// requests to the same url, making them external calls, so we can process in parallel
+// with multiple jRPC and Executor instances.
+//
+// the request flow will work as follows:
+// -> user do a trace batch request
+// -> jRPC balancer picks a jRPC server to handle the trace batch request
+// -> picked jRPC sends parallel trace transaction requests for each transaction in the batch
+// -> jRPC balancer sends each request to a different jRPC to handle the trace transaction requests
+// -> picked jRPC server group trace transaction responses from other jRPC servers
+// -> picked jRPC respond the initial request to the user with all the tx traces
+func (d *DebugEndpoints) TraceBatchByNumber(httpRequest *http.Request, number types.BatchNumber, cfg *traceConfig) (interface{}, types.Error) {
+	// timeout is the maximum time the code will wait for all the
+	// traces to be loaded from the jRPC and added to the responses
+	const timeout = 10 * time.Minute
+
+	// the size of the buffer defines
+	// how many txs it will process in parallel.
+	const bufferSize = 10
+
+	// builds the url of the jRPC server from the data found in the httpRequest
+	scheme := "http"
+	if httpRequest.URL.Scheme != "" {
+		scheme = httpRequest.URL.Scheme
+	}
+	u := url.URL{
```

Comment:
We must ensure the URL we get here is not from behind load balancer.

Reply:
we have validated and it is the external URL